### ISSUE TITLE
UX bugging on changing scene 

### ISSUE THREAD

**Issue by Ardoos** [Unity-Technologies/arfoundation-demos#37]
UX manager is bugging when I'm swapping between different scenes that use AR. My use is simple, first goal is to detect at least one plane and second one is to tap to place the object. When I finish one scene and load next one I don't have this part of UI (bottom part is working ok).

I modified Update in UIManager.cs with Debug to check what is the problem. 

```
    void Update ()
    {
        if (m_AnimationManager.localizeText)
        {
            if (!m_LocalizationManager.localizationComplete)
            {
                return;
            }
        }

        if (m_UXOrderedQueue.Count > 0 && !m_ProcessingInstructions)
        {
            // pop off
            m_CurrentHandle = m_UXOrderedQueue.Dequeue ();

            // fade on
            FadeOnInstructionalUI (m_CurrentHandle.InstructionalUI);
            m_GoalReached = GetGoal (m_CurrentHandle.Goal);
            m_ProcessingInstructions = true;
            m_FadedOff = false;
            Debug.Log ("Fade on " + m_CurrentHandle.Goal);
        }

        if (m_ProcessingInstructions)
        {
            // start listening for goal reached
            if (m_GoalReached.Invoke ())
            {
                // if goal reached, fade off
                if (!m_FadedOff)
                {
                    m_FadedOff = true;
                    m_AnimationManager.FadeOffCurrentUI ();
                    Debug.Log ("Fade off " + m_CurrentHandle.Goal);
                }
            }
        }
    }
```

When I change the scene, I call arsession.Reset() from another script, but it seems that phone is relocating all planes almost instantly. Therefore in console I have debug (based on Update() modifications) like this:
- Fade on FoundAPlane
- Fade off FoundAPlane
- Fade on PlacedAnObject
This 3 things happens almost at the same time. My app is working perfectly fine, but it turns out that ARUXAnimationManager class can't cope with this. It manages to change player to "tap to place" video, but is stuck on AlphaWhite therefore user can't see instruction.

How can I fix it so user instantly see "Tap to place object" instruction if phone already loaded planes on scene swap?

**Comment by DanMillerDev** [comment]
thanks for the report, I'll look into this and see if there's a fix to the code or if I can suggest a different workflow.

**Comment by DanMillerDev** [comment]
I was able to reproduce this in a similar environment and conditions and noticed the same behavior you did. The 'Tap To Place' video was playing but it was transparent and after placing an object it would flash on and fade off.

I've fixed this by adding an early check to the condition int his commit https://github.com/Unity-Technologies/arfoundation-demos/commit/4f5f889d50d301195afebc502022483dcb8d2695

This should hopefully fix the issue you were running into. I'd also recommend checking out the API for adding and showing custom UI at different times as it sounds like you have a bit more of a unique user flow in your app.
